Lib9c/Action/Sweep.cs

```diff
@@ -0,0 +1,263 @@
+                    monsterMap.Add(new KeyValuePair<int, int>(monster.CharacterId, monster.Count));
+                }
+            }
+
+            avatarState.questList.UpdateMonsterQuest(monsterMap);
+        }
+
+        private static void UpdateInventory(AvatarState avatarState, List<ItemBase> rewards)
+        {
+            var itemMap = new CollectionMap();
+            foreach (var reward in rewards)
+            {
+                itemMap.Add(avatarState.inventory.AddItem(reward));
+            }
+            avatarState.questList.UpdateCollectQuest(itemMap);
+        }
+
+        private void UpdateExp(AvatarState avatarState, int level, long exp)
+        {
+            var levelUpCount = level - avatarState.level;
+            var eventMap = new CollectionMap
+                { new KeyValuePair<int, int>((int)QuestEventType.Level, levelUpCount) };
+            avatarState.level = level;
+            avatarState.exp = exp;
+            avatarState.questList.UpdateCompletedQuest(eventMap);
+        }
+
+        public static List<ItemBase> GetRewardItems(IRandom random, int playCount,
+            StageSheet.Row stageRow,
+            MaterialItemSheet materialItemSheet)
+        {
+            var rewardItems = new List<ItemBase>();
+            var maxCount = random.Next(stageRow.DropItemMin, stageRow.DropItemMax + 1);
+            for (var i = 0; i < playCount; i++)
+            {
+                var selector = StageSimulator.SetItemSelector(stageRow, random);
+                var rewards = Simulator.SetRewardV2(selector, maxCount, random,
+                    materialItemSheet);
+                rewardItems.AddRange(rewards);
+            }
+
+            rewardItems = rewardItems.OrderBy(x => x.Id).ToList();
+            return rewardItems;
+        }
+
+        public static (int, long) GetLevelAndExp(CharacterLevelSheet characterLevelSheet,
+            AvatarState avatarState, int stageId, int repeatCount)
+        {
+            var remainCount = repeatCount;
+            var currentLevel = avatarState.level;
+            var currentExp = avatarState.exp;
+            while (remainCount > 0)
+            {
+                characterLevelSheet.TryGetValue(currentLevel, out var row, true);
+                var maxExp = row.Exp + row.ExpNeed;
+                var remainExp = maxExp - currentExp;
+                var stageExp = StageRewardExpHelper.GetExp(currentLevel, stageId);
+                var requiredCount = (int)Math.Ceiling(remainExp / (double)stageExp);
+                if (remainCount - requiredCount > 0) // level up
+                {
+                    currentExp += stageExp * requiredCount;
+                    remainCount -= requiredCount;
+                    currentLevel += 1;
+                }
+                else
+                {
+                    currentExp += stageExp * remainCount;
+                    break;
+                }
+            }
+
+            return (currentLevel, currentExp);
```

Comment:
Moved functions to `[AvatarStateExtensions](https://github.com/planetarium/lib9c/pull/921/commits/46aa8f4b4a2f81b83a14646f34a635c366d08f95#diff-8df980116a523e3c3f2215ca52307764a25a447f080cdd39f5c9dde9092be2a7R13)` class except `GetRewardItems()`. Because `GetRewardItems` are used only to sweep action.